pallets/unified-accounts/src/lib.rs

```diff
@@ -109,13 +112,18 @@ pub mod pallet {
         /// The overarching event type
         type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
         /// The Currency for managing evm address assets
-        type Currency: Mutate<Self::AccountId>;
+        type Currency: FungibleMutate<Self::AccountId>;
         /// Default evm address to account id conversion
         type DefaultEvmToNative: AddressMapping<Self::AccountId>;
         /// Default account id to evm address conversion
         type DefaultNativeToEvm: AccountMapping<Self::AccountId>;
         /// EVM chain id
+        #[pallet::constant]
         type ChainId: Get<u64>;
+        /// The amount of currency needed per mapping to be added.
+        /// TODO: calculate total bytes key + value in storage for mappings
+        #[pallet::constant]
+        type AccountMappingStorageFee: Get<BalanceOf<Self>>;
```

Comment:
What'll be the total bytes stored in storage mappings for this?
- `EvmToNative`:
     - value = 32 bytes
     - key = 32 (storage_prefix) + 16+20(blake2_128 concat) = 68 bytes
     - total = 100 bytes
- `NativeToEvm`:
     - value = 20 bytes
     - key = 32 + 16+32 = 80 bytes
     - total = 100 bytes

Using deposit function should fee = `deposit(2, 200)`?

Reply:
You don't need to calculate it like that. The assumption with storage fee calculation function is that key length is already accounted for in the entry price.

So it shold be `deposit(2, 32 + 20)`.